src/pkg/packager/prepare.go

```diff
@@ -193,6 +205,15 @@ func (p *Packager) FindImages() (imgMap map[string][]string, err error) {
 				message.Debugf("%s", contentString)
 				yamls, _ := utils.SplitYAML(contents)
 				resources = append(resources, yamls...)
+
+				// Check if the --why flag is set and if it is process the manifests
+				if whyImage != "" {
+					whyResourcesManifest, err := findWhyResources(resources, whyImage, component.Name, manifest.Name, false)
```

Comment:
```suggestion
					whyResourcesManifest, err := findWhyResources(yamls, whyImage, component.Name, manifest.Name, false)
```

Reply:
These should use the yamls from the current manfiest set (or chart for the above) - if `resources` is used then it will potentially repeat the resource into places that it actually doesn't exist (this shows in the repeated `podinfo` curlpods in the test and also in the `wordpress` example:
![image](https://github.com/defenseunicorns/zarf/assets/3977569/22dc8d9a-6dab-46f8-a6ae-b7da5270c09c)
...
![image](https://github.com/defenseunicorns/zarf/assets/3977569/56ce0360-9709-439e-97df-6d1da7249e6f)

(`connect-services` doesn't have a deployment in it)

Reply:
@Racer159 good catch!